odra-casper/livenet-env/src/livenet_host.rs

```diff
@@ -207,5 +249,57 @@ impl HostContext for LivenetHost {
         let timestamp = Timestamp::now();
         let client = self.casper_client.borrow_mut();
         rt.block_on(async { client.transfer(to, amount, timestamp).await })
+            .map_err(|e| {
+                self.map_error_code_to_odra_error(
+                    ContractId::Address(client.caller()),
+                    &e.error_message()
+                )
+            })
+    }
+
+    fn get_validator(&self, index: usize) -> PublicKey {
+        let rt = Runtime::new().unwrap();
+        let client = self.casper_client.borrow_mut();
+        rt.block_on(async { client.get_validator(index).await })
+    }
+
+    fn advance_with_auctions(&self, diff: u64) {
+        println!("advance_with_auctions called - Waiting for {diff} ms");
+        sleep(std::time::Duration::from_millis(diff));
+    }
+
+    fn auction_delay(&self) -> u64 {
+        let rt = Runtime::new().unwrap();
```

Comment:
Isn't there a better way of of calling async methods here?
Maybe reuse Runtime?
At least make an issue, that says to reimplement it.

Reply:
#544